File: sca/runtime/core/src/tuscany/sca/model/Service.cpp
/*
 *
 *  Copyright 2005 The Apache Software Foundation or its licensors, as applicable.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

/* $Rev$ $Date: 2005/12/22 11:33:21 $ */

#include "tuscany/sca/util/Logging.h"
#include "tuscany/sca/util/Exceptions.h"
#include "tuscany/sca/model/Service.h"

namespace tuscany
{
    namespace sca
    {
        namespace model
        {
            
            // Constructor
            Service::Service(Component* component, ServiceType* serviceType)
                : component(component), type(serviceType), binding(0)
            {
            }
            
            // Destructor
            Service::~Service()
            {
            }
            
            void Service::setBinding(ServiceBinding* binding)
            {
                this->binding = binding;
            }
            
            ServiceBinding* Service::getBinding() const
            {
                return binding;
            }
            
       } // End namespace model
    } // End namespace sca
} // End namespace tuscany

File: modules/java/org/apache/tuscany/Service.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 * 
 *   http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.    
 */

package org.apache.tuscany;

/**
 * Interface used to represent SCA component references providing both REST
 * access to a resource and function application.
 */
public interface Service {
    
    /**
     * Post a new item to a collection of items.
     */
    Iterable<String> post(Iterable<String> collection, Iterable<?> item);
    
    /**
     * Return an item.
     */
    Iterable<?> get(Iterable<String> id);
    
    /**
     * Update an item.
     */
    boolean put(Iterable<String> id, Iterable<?> item);
    
    /**
     * Delete an item.
     */
    boolean delete(Iterable<String> id);
    
    /**
     * Evaluate an expression.
     */
    <T> T eval(Object... params);
    
}
